src/services/db/GitFileSystemService.ts

```diff
@@ -433,9 +430,98 @@ export default class GitFileSystemService {
     })
   }
 
-  // TODO: Creates either directory or file
-  // ResourceDirectoryService used this to create a directory + file at the same time
-  create() {}
+  // Creates a file and the associated directory if it doesn't exist
+  create(
+    repoName: string,
+    userId: string,
+    content: string,
+    directoryName: string,
+    fileName: string
+  ): ResultAsync<
+    GitCommitResult,
+    ConflictError | GitFileSystemError | NotFoundError
+  > {
+    const filePath = directoryName ? `${directoryName}/${fileName}` : fileName
+    const pathToEfsDir = `${EFS_VOL_PATH}/${repoName}/${directoryName}/`
+    const pathToEfsFile = `${EFS_VOL_PATH}/${repoName}/${filePath}`
+    const encodedContent = content
+    let oldStateSha = ""
+
+    return this.getLatestCommitOfBranch(repoName, BRANCH_REF)
+      .andThen((latestCommit) => {
+        const { sha } = latestCommit
+        if (!sha) {
+          return errAsync(new GitFileSystemError("An unknown error occurred"))
+        }
+        oldStateSha = sha
+        return okAsync(true)
+      })
+      .andThen(() => this.getFilePathStats(repoName, directoryName))
+      .andThen(() => ok(""))
```

Comment:
what is this empty string for, seems like a redundant step?

Reply:
Whoops it's meant to convert stats into a boolean value but the functionality wasn't implemented fully, this has been modified in https://github.com/isomerpages/isomercms-backend/pull/892/commits/b3d87e344ceb2d146ceaf661b40c9548dcc5102d